### PR TITLE
Parser

### PR DESCRIPTION
**Pull request by paulvazbad**
# Parser v 1.0

Introducing the first version of the parser for the competition.
The Natural Language Processing Unit is located outside the ROS project in the RASA stack, however that information is retrieved form the parser.py by an api call.

Current intents defined:
- bring_something
- greet
- thank
- goodbye

## Further improvements to be made:

- Interpret intents into concrete actions for the robot, define: will that be done in rasa using an action server? or inside ros using a simple mapping of intent -> action . 
- Add more user interactions to the intent list, start an activity, go to place, chitchat, etc...
- Connect with existing audio/speech_to_text features in the project (#8 )

### CODE REVIEW

File: catkin_home/src/devices/CMakeLists.txt [hunk 1]
```diff
@@ -74,6 +74,7 @@ find_package(catkin REQUIRED COMPONENTS
    DEPENDENCIES
    std_msgs  # Or other packages containing msgs
    audio_common_msgs
+   intercom
```
Comment (diegocardozo97) [inline catkin_home/src/devices/CMakeLists.txt]: Is it necessary to include `intercom` in the `find_package()`?

File: catkin_home/src/action_selectors/scripts/parser.py [hunk 2]
```diff
@@ -78,6 +87,7 @@ def callback(msg):
     action_code.args = args
     rospy.loginfo(action_code)
     pub.publish(action_code)
+    
 
 def listener():
```
Comment (diegocardozo97) [inline catkin_home/src/action_selectors/scripts/parser.py]: This could be called `main()` or something else.

File: catkin_home/src/action_selectors/scripts/parser.py [hunk 3]
```diff
@@ -64,9 +71,11 @@ def callback(msg):
     #open list of actions
     #call parser
     #send message to engine 
+    pub_resp = rospy.Publisher('BotResponse', response, queue_size=10)
```
Comment (diegocardozo97) [inline catkin_home/src/action_selectors/scripts/parser.py]: Is creating this again and again in each callback okay?

File: rasa/runrasa [hunk 4]
```diff
@@ -0,0 +1,3 @@
+#!/bin/env bash
```
Comment (diegocardozo97) [inline rasa/runrasa]: Do we need also this file?

File: catkin_home/src/devices/src/InputAudio.cpp [hunk 5]
```diff
@@ -9,6 +9,7 @@ void onAudioCallback(const audio_common_msgs::AudioData::ConstPtr msg){
     cout<<"I heard "<<endl;
     vector<uint8_t> test;
     test.push_back(1);
+   //Define otro mensaje que va a enviar
```
Comment (diegocardozo97) [inline catkin_home/src/devices/src/InputAudio.cpp]: Please use the next time the word "TODO:" the next time if this is a to-do. Also, I think the call of the parser will be in action_selectors/hear.py instead.

File: catkin_home/src/devices/CMakeLists.txt (hunk 1 above)
Comment (paulvazbad) [reply catkin_home/src/devices/CMakeLists.txt]: it is! , I removed previously the connection with main_engine and forgot about it, thanks for pointing it out.

File: rasa/tts.py [hunk 6]
```diff
@@ -0,0 +1,20 @@
+import requests
```
Comment (diegocardozo97) [inline rasa/tts.py]: Is this file still needed?

**Comment by diegocardozo97** [comment]
Please review if the file rasa/out.log is necessary. 
Also, maybe next time we can include a text file mentioning the versions and reasons of the files in rasa/models.

Also, please take care in the "devices" package when merging, because this seems to have the old version before #8.

File: rasa/tts.py (hunk 6 above)
Comment (paulvazbad) [reply rasa/tts.py]: For future debugging purposes it might be useful.

File: rasa/runrasa (hunk 4 above)
Comment (paulvazbad) [reply rasa/runrasa]: I guess it is simpler to type ./runrasa than "rasa run -m models --enable-api --log-file out.log" and if in the future we include an action_server to the rasa module we can include it to run in the bash script.

File: catkin_home/src/action_selectors/scripts/parser.py (hunk 2 above)
Comment (paulvazbad) [reply catkin_home/src/action_selectors/scripts/parser.py]: you mean def listener() change it to main() ?

File: catkin_home/src/action_selectors/scripts/parser.py (hunk 3 above)
Comment (paulvazbad) [reply catkin_home/src/action_selectors/scripts/parser.py]: It is not, i will fix it.

File: catkin_home/src/devices/src/InputAudio.cpp (hunk 5 above)
Comment (paulvazbad) [reply catkin_home/src/devices/src/InputAudio.cpp]: okok this file is outdated though so the new version found in the develop branch will be kept instead.

File: catkin_home/src/action_selectors/scripts/parser.py (hunk 2 above)
Comment (diegocardozo97) [reply catkin_home/src/action_selectors/scripts/parser.py]: Yes

File: rasa/runrasa (hunk 4 above)
Comment (diegocardozo97) [reply rasa/runrasa]: Oh, I mean that there is also the file rasa/runrasa.bash with exactly the same.

Comment (paulvazbad) [reply rasa/runrasa]: ah lol thanks
